### ISSUE TITLE
920450: @within %{tx.restricted_headers} breaks range header

### ISSUE THREAD

**Issue by dune73** [SpiderLabs/owasp-modsecurity-crs#490]
PR #427 broke the range header by moving from a regex to a `@within` construct. Unfortunately, the underdocumented operator treats the hyphen as separator. 
Our new
`setvar:'tx.restricted_headers=proxy lock-token content-range translate if'"`
is thus a syntactical equivalent to 
`setvar:'tx.restricted_headers=proxy lock token content range translate if'"`

The range header will thus trigger an alert. I take it a unit-test is due. ;)

But how do we go about this? Revert back to regex in combination with this definition?
`setvar:'tx.restricted_headers=/proxy/ /lock-token/ /content-range/ /translate/ /if/'"`

**Comment by lifeforms** [comment]
Nice find! This is definitely a bug.

When checking the [source](https://github.com/SpiderLabs/ModSecurity/blob/v3/master/src/operators/within.cc), it appears that `@within` just does a substring match. So it's not that the dash has a special meaning, just *any* substring of `tx.restricted_headers` will match the rule, like a header called `Roxy` or `Oxy`.

I think it's fine to keep using `@within` but we should pre-process the header name like in CRS2, e.g.

```
SecRule REQUEST_HEADERS_NAMES "^(.*)$"
   capture,setvar:'tx.header_name=/%{tx.0}/'

SecRule TX:HEADER_NAME "@within %{tx.restricted_headers}"
```

And go back to the slash delimited format as you said.

When we change this, please ensure #463 is merged first to prevent conflicts in the conf file.

**Comment by dune73** [comment]
I did an inverse check and looked for a different substring, so this did not occur to me. Thanks for getting this right.

I'll provide a PR.

**Comment by lifeforms** [comment]
Fixed in #491, great work!
